api/handler/v1beta1/job_spec.go

```diff
@@ -163,30 +163,31 @@ func (sv *JobSpecServiceServer) CreateJobSpecification(ctx context.Context, req
 		return nil, mapToGRPCErr(sv.l, err, "unable to get namespace")
 	}
 
-	jobSpec, err := FromJobProto(req.GetSpec(), sv.pluginRepo)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
+	reqJobSpecs := req.GetSpecs()
+	var jobSpecs []models.JobSpec
+	for _, spec := range reqJobSpecs {
+		jobSpec, err := FromJobProto(spec, sv.pluginRepo)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
+		}
+		jobSpecs = append(jobSpecs, jobSpec)
 	}
 
-	// validate job spec
-	if err = sv.jobSvc.Check(ctx, namespaceSpec, []models.JobSpec{jobSpec}, logWriter); err != nil {
-		return nil, status.Errorf(codes.Internal, "spec validation failed\n%s", err.Error())
+	jobNames := make([]string, len(jobSpecs))
+	for i, jobSpec := range jobSpecs {
+		jobNames[i] = jobSpec.Name
 	}
+	deployID, err := sv.jobSvc.CreateAndDeploy(ctx, namespaceSpec, jobSpecs, logWriter)
```

Comment:
let's remove the space just below this line, so there should be no gap between this and the error check. also, you might want to add the failure case unit test for `CreateAndDeploy` to increase the coverage.

Reply:
if you can also add a unit test for deserializing job failure in this function, that would be great.

Reply:
nice idea, added test cases